nautobot_golden_config/utilities/config_postprocessing.py

```diff
@@ -0,0 +1,177 @@
+            f"check the template for invalid Jinja syntax.\n"
+        ) from error
+    except jinja_errors.TemplateError as error:  # Catches all remaining Jinja errors
+        raise RenderConfigToPushError(
+            "Jinja encountered an unexpected TemplateError; check the template for correctness\n"
+        ) from error
+
+
+def get_config_postprocessing(configs: models.GoldenConfig, request: HttpRequest) -> str:
+    """Renders final configuration  artifact from intended configuration.
+
+    It chains multiple callables to transform an intended configuration into a configuration that can be pushed.
+    Each callable should match the following signature:
+    `my_callable_function(config_postprocessing: str, configs: models.GoldenConfig, request: HttpRequest)`
+
+    Args:
+        configs (models.GoldenConfig): Golden Config object per device, to retrieve device info, and related configs.
+        request (HttpRequest): HTTP request for context.
+    """
+    if not ENABLE_POSTPROCESSING:
+        return (
+            "Generation of intended configurations postprocessing it is not enabled, check your plugin configuration."
+        )
+
+    config_postprocessing = configs.intended_config
+    if not config_postprocessing:
+        return (
+            "No intended configuration is available. Before rendering the configuration with postprocessing, "
+            "you need to generate the intended configuration."
+        )
+
+    # Available functions to create the final intended configuration, in string dotted format
+    # The order is important because, if not changed by the `config_postprocessing_subscribed`, is going
+    # to be used to process the intended configuration in this specific order
+    default_config_postprocessing_callables = [
+        "nautobot_golden_config.utilities.config_postprocessing.render_secrets",
+    ]
+    # The available methods can be extended by configuration settings from config_postprocessing_callables
+    config_postprocessing_callables = default_config_postprocessing_callables + PLUGIN_CFG.get(
+        "config_postprocessing_callables", []
+    )
+
+    # Subscribed callables to post-process the intended configuration, in a specific order. With this option, you could
+    # skip some default callables, such as `render_secrets` if not desired.
+    # In the future, this could be taken from query parameters.
+    config_postprocessing_subscribed = config_postprocessing_callables
+    if PLUGIN_CFG.get("config_postprocessing_subscribed"):
+        config_postprocessing_subscribed = PLUGIN_CFG["config_postprocessing_subscribed"]
+
+    for func_name_subscribed in config_postprocessing_subscribed:
+        if func_name_subscribed in config_postprocessing_callables:
+            try:
+                func = import_string(func_name_subscribed)
+            except ImportError as error:
+                raise ValueError(f"{func_name_subscribed} doesn't look a valid function. {error}") from error
```

Comment:
```suggestion
                        msg = (
                        "There was an issue attempting to import a `config_postprocessing_callables` function of"
                        f"{func_name_subscribed}, this is expected with a local configuration issue "
                       "and not related to the Golden Configuration Plugin, please contact your system admin for further details"
                )
                raise ValueError(msg + error) from error
```
I may not have the spacing right, so please review.

Reply:
resolved in https://github.com/nautobot/nautobot-plugin-golden-config/commit/5b88c8afa091542b049f6de527ecb0d4001df50b